confmap/internal/e2e/types_test.go

```diff
@@ -98,27 +98,22 @@ func TestTypeCasting(t *testing.T) {
 		{
 			value:       "\"0123\"",
 			targetField: TargetFieldString,
-			expected:    "0123",
-		},
-		{
-			value:       "\"0123\"",
-			targetField: TargetFieldInt,
-			expected:    83,
+			expected:    "\"0123\"",
```

Comment:
This breaks users right?
If I entered `endpoint: "http://localhost:4317"` instead of my exporter using `http://localhost:4317` would it attempt to use a value that contains double-quote literals?

Reply:
> If I entered endpoint: "http://localhost:4317" instead of my exporter using http://localhost:4317 would it attempt to use a value that contains double-quote literals?

That would happen if you did 

```
endpoint: ${env:ENDPOINT}
```

with `ENDPOINT` set to `"http://localhost:4317"` (quotes included). The changes only affects how the values passed via a provider are parsed